cli/coredump/coredump.go

```diff
@@ -0,0 +1,130 @@
+package coredump
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/codeclysm/extract"
+	"github.com/tarantool/tt/cli/cmdcontext"
+)
+
+//go:embed scripts/*
+var corescripts embed.FS
+
+func readEmbedFile(fs embed.FS, path string) (string, error) {
+	content, err := fs.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(content), nil
+}
+
+// packs coredump into a tar.gz archive.
+func pack(coreName string) error {
+	ctx, err := readEmbedFile(corescripts, "scripts/tarabart.sh")
+	if err != nil {
+		return fmt.Errorf("There was some problem packing archive. \n"+
+			"Error: '%v'.\n", err)
+	}
+	data := []byte(ctx)
+	path, err := os.Getwd()
+	err = ioutil.WriteFile(path+"/tarabart.sh", data, 0777)
+	defer os.Remove(path + "/tarabart.sh")
+	if err != nil {
+		return fmt.Errorf("There was some problem packing archive. \n"+
+			"Error: '%v'.\n", err)
+	}
+	cmd := exec.Command(path+"/tarabart.sh", "-c", path+"/"+coreName)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err == nil {
+		fmt.Printf("Core was succesfully packed. \n")
+	} else {
+		err = fmt.Errorf("There was some problem packing archive. \n"+
+			"Error: '%v'.\n", err)
+	}
+	return err
+}
+
+// unpacks a tar.gz archive.
+func unpack(tarName string) error {
+	path, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("There was some problem unpacking archive. \n"+
+			"Error: '%v'.\n", err)
+	}
+	data, err := ioutil.ReadFile(path + "/" + tarName)
+	if err != nil {
+		return fmt.Errorf("There was some problem unpacking archive. \n"+
+			"Error: '%v'.\n", err)
+	}
+	buffer := bytes.NewBuffer(data)
+	err = extract.Gz(context.Background(), buffer, path+"/", nil)
+	if err == nil {
+		fmt.Printf("Archive was succesfully unpacked. \n")
+	} else {
+		err = fmt.Errorf("There was some problem unpacking archive. \n"+
+			"Error: '%v'.\n", err)
+	}
+	return err
+}
+
+// inspect allows user to inspect unpacked coredump.
+func inspect(coreFolder string) error {
+	ctx, err := readEmbedFile(corescripts, "scripts/gdb.sh")
```

Comment:
Seems like the same code was written by three times. WET is not a good practice. I think the code can be wrapped in one function. And this function will be called from pack/unpack/inspect with different parameters.

Reply:
Fixed